task4/src/services/groupService.js

```diff
@@ -0,0 +1,71 @@
+export default class GroupService {
+    constructor(models) {
+        this.model = models.Group;
+        this.userModel = models.User;
+    }
+
+    getAllGroups() {
+        try {
+            const data = this.model.findAll({
+                include: [
+                    {
+                        model: this.userModel,
+                        as: 'users',
+                        required: false,
+                        attributes: ['id', 'login'],
+                        through: { attributes: [] }
+                    }
+                ]
+            });
+            return data;
+        } catch (err) {
+            throw err;
```

Comment:
BTW,  what will be caught here in catch without await?

Reply:
Is it possible that model's method can throw an error?